File: 1430-find-the-k-beauty-of-a-number/1430-find-the-k-beauty-of-a-number.java
class Solution {
    public int divisorSubstrings(int num, int k) {
        int res = 0;
        String numstr = String.valueOf(num);
        for(int i = 0 ; i <= numstr.length() - k  ; i++){
            int n = Integer.parseInt(numstr.substring(i , i+k));
            if(n !=  0 && num % n == 0){
                res += 1;
            }
        }
        return res;
    }
}

File: 1430-find-the-k-beauty-of-a-number/1430-find-the-k-beauty-of-a-number.py
class Solution:
    def divisorSubstrings(self, num: int, k: int) -> int:
        res = 0
        numstr = str(num)
        for i in range(len(numstr) - k + 1):
            if int(numstr[i:i+k]) != 0 and num % int(numstr[i:i+k]) == 0:
                res += 1
        return res